src/components/SaveGameForm.js

```diff
@@ -87,22 +89,70 @@ export default function SaveGameForm({onSubmit}) {
     function handleChange(event) {
         setFormInput({
             ...formInput,
-            [event.target.name]: event.target.value,
+            [event.target.name]: event.target.value
         })
     }
 
     function handleSubmit(event) {
         event.preventDefault()
-        onSubmit(formInput)
+        const validation = validateForm(formInput)
+        if (validation.result) {
+            trimInputs(formInput)
+            onSubmit(formInput)
+            setFormInput({
+                location: '',
+                date: '',
+                players: '',
+                winner: '',
+                shots:'',
+            })
+            setErrorMessage('')
+        } else {
+           setErrorMessage(validation.message)
+        }
+    }
+
+    function validateForm(formInput) {
+       const filledInputs = validateNotEmpty(formInput)
+       const validNumber = validateShots(formInput)
+
+       if (filledInputs.result) {
+           if (validNumber.result) {
+                return {result: true}
+           } else {
+               return validNumber
+           }
+       } else {
+           return filledInputs
+       }
+    }
+
+    function validateNotEmpty({location, date, players, winner, shots}) {
+       const allInputsFilled = location.trim().length > 0 && date.trim().length > 0 && players.trim().length > 0 && winner.trim().length > 0 && shots.trim() > 0
+       if (allInputsFilled) {
+            return {result: true}
```

Comment:
I think `result` is incorrect here, why not `isValid`? Even if the validation returns that the input is not valid, it is still a result :)

Reply:
Ah sure, because it is boolean. Sounds good. :)